### ISSUE TITLE
Accessing Sharepoint file after authentication

### ISSUE THREAD

**Issue by SM-138496** [vgrem/Office365-REST-Python-Client#704]
Hello, I'm having problems with getting files after successfully accessing Sharepoint. I try different approaches but I get the same error: 
b'{"error":{"code":"-2130575338, Microsoft.SharePoint.SPException","message":{"lang":"en-US","value":"The file /PATH/TO/FILE does not exist."}}}'

one example code would be: 

credentials = ClientCredential(x, y)
ctx = ClientContext(z).with_credentials(credentials)
web = ctx.web
ctx.load(web)
ctx.execute_query()
print("Web site title: {0}".format(web.properties['Title']))

response = File.open_binary(ctx, relative_url)
bytes_file_obj = io.BytesIO()
bytes_file_obj.write(response.content)
bytes_file_obj.seek(0)

print(bytes_file_obj.getvalue())

---------
when z = 'https://tenant.sharepoint.com/sites/test/' and relative_url = '/sites/test/Shared%20Documents/TestBook.xlsx'

It would print 'Web site title: test' 
but give b'{"error":{"code":"-2130575338, Microsoft.SharePoint.SPException","message":{"lang":"en-US","value":"The file /sites/test/Shared Documents/TestBook.xlsx does not exist."}}}'
after

I try using /Shared Documents/, /Documents/, /Shared%20Documents/
And try creating new test files on other locations like 'tenant.sharepoint.com/Shared Documents/TestBook.xlsx' or 'tenant.sharepoint.com/Shared Documents/Folder/TestBook.xlsx' but still, I got the same error

When I copy and paste the full URL on the browser (example: https://tenant.sharepoint.com/sites/test/Shared Documents/TestBook.xlsx) it shows a pop-up to download the file, so the directory exists. 

I basically ran out of ideas on how to solve that.

**Comment by guidorietbroek** [comment]
Did you try https://tenant.sharepoint.com:/sites/test/?

mind the “:” inside the url after .com

**Comment by SM-138496** [comment]
> Did you try https://tenant.sharepoint.com:/sites/test/?
> 
> mind the “:” inside the url after .com

Just test it, it didn't change the error (still get the site's title but it said that the file does not exist)

**Comment by SM-138496** [comment]
Anyone else has an idea what this can be?

**Comment by dkuska** [comment]
Are you able to get the file like this and look at the properties of it?
```python
file = ctx.web.get_file_by_server_relative_url(relative_url).get().execute_query()
```

**Comment by SM-138496** [comment]
> Are you able to get the file like this and look at the properties of it?
> 
> ```python
> file = ctx.web.get_file_by_server_relative_url(relative_url).get().execute_query()
> ```

nope, it gives me this message:
ClientRequestException: ('-2130575338, Microsoft.SharePoint.SPException', 'The file /sites/test/Shared%20Documents/TestBook.xlsx does not exist.', "404 Client Error: Not Found for url: https://[tenant].sharepoint.com/sites/test/_api/Web/getFileByServerRelativeUrl('%2Fsites%2Ftest%2FShared%2520Documents%2FTestBook.xlsx')")

**Comment by SM-138496** [comment]
> Ahhh, a clue! You shouldn't manually escape the space character, as the `%` itself gets escaped and then the url does not make sense

Sorry I didn't understand, I tried using /Shared Documents/ and /Shared%20Documents/ both didn't work ( just tried /SharedDocuments/ without space and it got the same error)

**Comment by dkuska** [comment]
The library takes care of escaping the space character into `%20`.

Your syntax looks correct so I'm suspecting a permission issue.
You could also try if you can access the `/Shared Documents/` folder itself, but I suspect that this also would not work.

**Comment by SM-138496** [comment]
> The library takes care of escaping the space character into `%20`.
> 
> Your syntax looks correct so I'm suspecting a permission issue. You could also try if you can access the `/Shared Documents/` folder itself, but I suspect that this also would not work.

Yeah, I tried it before but double-check now, it said Shared Documents does not exist.
As for permission, I gave something similar to this on sharepoint tenant and sharepoint test site respectively: 
	<AppPermissionRequests AllowAppOnlyPolicy="true">
	  <AppPermissionRequest Scope="http://sharepoint/content/tenant" Right="FullControl" />
	</AppPermissionRequests>
&
	<AppPermissionRequests AllowAppOnlyPolicy="true">
 	 <AppPermissionRequest Scope="http://sharepoint/content/tenant/sites/test" Right="FullControl" />
	</AppPermissionRequests>

Before if I did something wrong on the permission part I would not even get the title; Now because it is just giving an error with the files and not the web.properties part I thought it was not related to the XML permissions...

**Comment by SM-138496** [comment]
Found it. When I set the permissions:
```
<AppPermissionRequests AllowAppOnlyPolicy="true">
 <AppPermissionRequest Scope="http://sharepoint/content/tenant" Right="FullControl" />
</AppPermissionRequests>
```
I replace 'tenant' with the company name just like you do with things like the URL. If I made the scope literally 'http://sharepoint/content/tenant' or "http://sharepoint/content/sitecollection" (on the specific site depending on the user case). The script works fine.

One interesting note is that /Shared%20Documents/ and /Shared Documents/ works.

The only thing I don't understand is that normally when I mistake something on the permissions I cannot access the web.properties as I get an error related to authentication. But in this case, I do get web.properties and it says the files do not exist, which is weird.

 Anyway, it is finally working now 👍
